[calligra] Work around QLocale().timeFormat().simplified() seg faulting in Qt5.6.

dcaliste opened the pull request:
This is an ugly workaround to avoid segmentation fault when opening a spreadsheet in sailfish-office with Qt5.6.

The ```-std=c++0x``` flag make the following code segmentation fault:
<pre>
#include &lt;QLocale>
#include &lt;stdio.h>

int main(int argc, char ** argv)
{
  fprintf(stderr, "'%s'\n", QLocale().timeFormat().simplified().toLocal8Bit().data());
  return 0;
}
</pre>

Maybe this code is illegal for C++ standard, I don't know. It's working well without the flag though.

rburchell commented:
Do you have a backtrace of the crash? (both for your testcase, and for the Calligra patch you're applying).

Does your testcase still crash if you add a QCoreApplication or QGuiApplication instance?

dcaliste commented:
Here is the backtrace when compiled with -std=c++0x. It's working ok without this flag.

<pre>
Program received signal SIGSEGV, Segmentation fault.
0xb6d53852 in QString::simplified_helper(QString&) () from /usr/lib/libQt5Core.so.5
(gdb) bt
#0  0xb6d53852 in QString::simplified_helper(QString&) () from /usr/lib/libQt5Core.so.5
#1  0x00010c38 in _ZNO7QString10simplifiedEv ()
#2  0x000109c0 in main ()
</pre>

dcaliste commented:
The test program is still crashing when adding a QApplication and compiled with -std=c++0x.

<pre>
#include &lt;QLocale>
#include &lt;QApplication>
#include &lt;stdio.h>

int main(int argc, char ** argv)
{
  QApplication app(argc, argv);

  fprintf(stderr, "'%s'\n", QLocale().timeFormat().simplified().toLocal8Bit().data());
  return 0;
}
</pre>

If QLocale().timeFormat() is copied into another QString before calling simplified(), it's working. Using ```QStringLiteral("HH:mm:ss t").simplified()``` is also working.

I guess the issue is in Qt, but I don't know where to look for. I didn't find anything usefull on the Internet neither…

pvuorela commented:
Could be even a compiler error since it triggers with such flag? QString seems internally using some template code on simplified_helper.

I'm a bit wary of making behavioral changes here. If that copying into another QString works, maybe it could be used instead as workaround?

dcaliste commented:
You're right, maybe a compiler error. I wanted to avoid any other places where this may happen with the workaround, but the copying is a better solution. Will implement and test tomorrow. Thanks for your reply.

dcaliste commented:
Here the new workaround, less ugly, and keeping the functionalities 1:1.

I've greped in the all sources to find another ```locale.*simplified()``` pattern, but it was the only place.

pvuorela commented:
Simple enough for a workaround. Let's just have this for now.

pvuorela commented:
And thanks!

dcaliste commented:
You're welcome, I'm happy it's working again (I'm receiving my orchestra playing schedules as Excel sheets…).
